hip-clang.docker

```diff
@@ -28,6 +28,7 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-
     libpthread-stubs0-dev \
     llvm-amdgpu \
     miopengemm \
+    ninja-build \
```

Comment:
@pfultz2 TODO: remove Ninja build once https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/43 closed.

Reply:
Addressed in aef083f7dd5f1d98c4132371350bf594f98c64bd